### PR TITLE
Hidden participants

### PR DESCRIPTION
**Pull request by frostbyte73**

### CODE REVIEW

File: pkg/rtc/room.go [hunk 1]
```diff
@@ -296,9 +296,17 @@ func (r *Room) CloseIfEmpty() {
 
 	r.lock.RLock()
 	numParticipants := len(r.participants)
+	if numParticipants == 1 {
```
Comment (davidzhao) [inline pkg/rtc/room.go:299]: what if there are multiple participants here?

File: pkg/rtc/room.go [hunk 2]
```diff
@@ -472,6 +480,10 @@ func (r *Room) subscribeToExistingTracks(p types.Participant) {
 
 // broadcast an update about participant p
 func (r *Room) broadcastParticipantState(p types.Participant, skipSource bool) {
+	if p.Hidden() {
```
Comment (davidzhao) [inline pkg/rtc/room.go:483]: I'm trying to think if there's ever a case the hidden participant should get updates about itself... I think the only thing is metadata updates.. if someone updated that participant's metadata on the server.

it seems that it's more future-proof if this function sent that through.

File: pkg/service/localroomstore.go [hunk 3]
```diff
@@ -119,7 +119,7 @@ func (p *LocalRoomStore) GetParticipant(roomName, identity string) (*livekit.Par
 	return participant, nil
 }
 
-func (p *LocalRoomStore) ListParticipants(roomName string) ([]*livekit.ParticipantInfo, error) {
+func (p *LocalRoomStore) ListParticipants(roomName string, includeHidden bool) ([]*livekit.ParticipantInfo, error) {
```
Comment (davidzhao) [inline pkg/service/localroomstore.go:122]: IMO, we should not hide hidden participants from room APIs.. since they are used by admins of the room

**Review by davidzhao — COMMENTED** [review]
looks good! it'd be great to see test coverage for this in room_test.

File: pkg/rtc/room.go (hunk 1 above)
Comment (frostbyte73) [reply pkg/rtc/room.go:299]: then we won't close the room

Comment (davidzhao) [reply pkg/rtc/room.go:299]: we should close the room as long as there are no visible participants left.

**Review by davidzhao — APPROVED** [review]
awesome! looks great.

**Pull request merged**